pulp_smash/tests/rpm/cli/test_upload.py

```diff
@@ -0,0 +1,114 @@
+    def setUpClass(cls):
+        """Create a repository and download temporary drpm."""
+        super(UploadDrpmTestCase, cls).setUpClass()
+
+        if inspect.getmro(cls)[0] == UploadDrpmTestCase:
+            raise unittest2.SkipTest('Abstract base class.')
+        cls.cfg = config.get_config()
+        cls.repo_id = utils.uuid4()
+        cli.Client(cls.cfg).run(
+            'pulp-admin rpm repo create --repo-id {}'
+            .format(cls.repo_id).split()
+        )
+
+        completed_process = cli.Client(cls.cfg).run(
+            'mktemp --directory'.split()
+        )
+        cls.temp_folder = completed_process.stdout.strip()
+        cls.temp_drpm = os.path.join(cls.temp_folder, DRPM)
+
+        cli.Client(cls.cfg).run(
+            'curl -o {} {}'
+            .format(cls.temp_drpm, DRPM_URL).split()
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete the repository and created by :meth:`setUpClass`."""
+        super(UploadDrpmTestCase).tearDownClass()
+
+        cli.Client(cls.cfg).run(
+            'pulp-admin rpm repo delete --repo-id {}'
+            .format(cls.repo_id).split()
+        )
+        cli.Client(cls.cfg).run('pulp-admin orphan remove --all'.split())
+        cli.Client(cls.cfg).run('rm -rf {}'.format(cls.temp_folder).split())
+
+    def test_upload_and_output(self):
+        """Test upload of DRPM.
+
+        Steps
+
+        1. Upload DRPM
+        2. Check output of ``pulp-admin rpm repo content drpm``
+
+        https://github.com/PulpQE/pulp-smash/issues/336
+        """
+        cli.Client(self.cfg).run(
+            'pulp-admin rpm repo uploads drpm --repo-id {} -f {}'
+            .format(self.repo_id, self.temp_drpm).split()
+        )
+        completed_process = cli.Client(self.cfg).run(
+            'pulp-admin rpm repo content drpm --repo-id {}'
+            .format(self.repo_id).split()
+        )
+
+        excepted_output = ('\x1b[0m'
+                           'Arch:         noarch\n'
+                           'Checksum:     e348519427b367a13f1898f3fc02235acb7'
+                           '56cfdb834b3ca391416216b423a9e\n'
+                           'Checksumtype: sha256\n'
+                           'Epoch:        0\n'
+                           'Filename:     drpms/yum-3.4.3-10.fc16_3.4.3-11.fc'
+                           '16.noarch.drpm\n'
+                           'Release:      11.fc16\n'
+                           'Version:      3.4.3\n'
+                           '\n\n')
+
+        self.assertEqual(completed_process.stdout, excepted_output)
+
+    def test_upload_skip_existing(self):
+        """Test duplicit upload of DRPM with flag ``--skip-existing``.
+
+        Steps
+
+        1. Upload DRPM
+        2. Upload again with flag and check if it wasn't uploaded.
+
+        https://github.com/PulpQE/pulp-smash/issues/336
+        """
+        cli.Client(self.cfg).run(
+            'pulp-admin rpm repo uploads drpm --repo-id {} -f {}'
```

Comment:
What does `-f` do? I'd recommend expanding this to the full option name, either in this commit or in the reviewer's follow-up commit.